### PR TITLE
Add Speculation Rules S&P Questionnaire.

### PR DESCRIPTION
**Pull request by jeremyroman**

### CODE REVIEW

File: speculation-rules-security-privacy-questionnaire.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+At its core the feature does not expose new information to web sites or other parties. However, the user agent may expose information as a consequence of use of this feature.
+
+The feature allows the user agent to use its discretion about which outgoing hypertext references should be prefetched. While we anticipate that user agents will take action to [mitigate user tracking risks](browsing-context.md#privacy-based-restrictions) for cross-origin prefetches, including by denying access to cookies and storage, it remains true that the origin server that serves a prefetch may learn information such as:
+* the request URL
+* the referrer origin from which the prefetch was initiated
+* any information remaining in the request headers which may identify the user agent
+* the client IP address, if it is not anonymized
+
+This is necessary in order to issue an HTTP request to fetch the resource(s).
```
Comment (domenic) [inline speculation-rules-security-privacy-questionnaire.md:15]: "Note that all other fetches on the web platform, e.g. using the `fetch()` API, expose the same information" maybe? You say that below in the general sense, but IMO these specific bullet points can be a bit scary until you actually take a moment to think about them.

File: speculation-rules-security-privacy-questionnaire.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+This is substantially similar to the answer to the first question.
+
+The prefetches/prerenders themselves may include PII if a site includes them in the requested URL (as with any outgoing fetch or navigation).
+
+The fact that a prefetch or prerender occurs is visible to its origin server. Accordingly, user agents should avoid basing this decision on personal information not already known to the site. For example, if a user agent prefetched a link based on a profile of the user's interests from their browsing activity across the web, the fact that a prefetch was executed may reveal information about those interests that would not have otherwise been exposed.
+
+To the extent that the user agent modifies the path taken by network traffic in order to accommodate a request to anonymize the client IP address, encrypted traffic that may be derived from PII may be visible to parties along the revised network path, including the proxy operator To mitigate this, traffic should be encrypted with TLS and not include cookies or other identifiers.
```
Comment (domenic) [inline speculation-rules-security-privacy-questionnaire.md:39]:
```suggestion
To the extent that the user agent modifies the path taken by network traffic in order to accommodate a request to anonymize the client IP address, encrypted traffic that may be derived from PII may be visible to parties along the revised network path, including the proxy operator. To mitigate this, traffic should be encrypted with TLS and not include cookies or other identifiers.
```

Comment (domenic) [reply speculation-rules-security-privacy-questionnaire.md:39]: "...as is [required](https://github.com/jeremyroman/alternate-loading-modes/blob/main/fetch.md#fetching-with-no-credentials) by other parts of the proposal" maybe?

**Review by domenic — APPROVED** [review]

File: speculation-rules-security-privacy-questionnaire.md (hunk 1 above)
Comment (jeremyroman) [reply speculation-rules-security-privacy-questionnaire.md:15]: Done.

File: speculation-rules-security-privacy-questionnaire.md (hunk 2 above)
Comment (jeremyroman) [reply speculation-rules-security-privacy-questionnaire.md:39]: Compromised and used that link on the phrase "not include cookies or other identifiers" in the interest of brevity; hopefully that captures it.

**Pull request merged**